src/CMakeLists.txt

```diff
@@ -1,116 +1,88 @@
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3ai38.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3ai39.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3aq15.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3as00.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3difdat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3doxdat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi01.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi04.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi58.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi59.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi62.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi63.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi64.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi68.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi69.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi71.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi72.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi73.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi74.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi75.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi76.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi82.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi83.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi88.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fi92.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fp11.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fp12.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fp13.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fs13.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fs15.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fs21.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3fs26.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3ft32.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3kind.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3locdat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3movdat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3nogds.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3pradat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3reddat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3tagb.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3trnarg.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3unpk77.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3utcdat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3valdat.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/w3ymdh4.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/xmovex.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/xstore.f
-  PARENT_SCOPE)
 
-set(c_src
-  ${CMAKE_CURRENT_SOURCE_DIR}/mova2i.c
-  ${CMAKE_CURRENT_SOURCE_DIR}/summary.c
-  PARENT_SCOPE)
+include("list_of_files.cmake")
+
+if(APPLE)
+  add_compile_definitions(APPLE)
+elseif(UNIX)
+  add_compile_definitions(LINUX)
+endif()
+
+if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_C_FLAGS "-g")
+  set(CMAKE_C_FLAGS_RELEASE "-O3")
+  set(CMAKE_Fortran_FLAGS "-g")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+  set(fortran_d_flags "-r8")
+  set(fortran_8_flags "-i8 -r8")
+elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_C_FLAGS "-ggdb")
+  set(CMAKE_C_FLAGS_RELEASE "-O3")
+  set(CMAKE_Fortran_FLAGS "-ggdb -fno-range-check -fbacktrace -funroll-loops")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+  set(fortran_d_flags "-fdefault-real-8")
+  set(fortran_8_flags "-fdefault-integer-8 -fdefault-real-8")
+endif()
+
+set_source_files_properties(${c_src} PROPERTIES COMPILE_OPTIONS "${c_flags}")
+set_source_files_properties(${c_src} PROPERTIES COMPILE_DEFINITIONS
+  "${platform_definitions}")
```

Comment:
variable `platform_definitions` is not defined and hence this line does nothing.
please remove.